demo/src/main/scala/dev/atedeg/ecscalademo/systems/BallCreationSystem.scala

```diff
@@ -0,0 +1,33 @@
+package dev.atedeg.ecscalademo.systems
+
+import dev.atedeg.ecscala.*
+import dev.atedeg.ecscala.dsl.ECScalaDSL
+import dev.atedeg.ecscala.util.types.given
+import dev.atedeg.ecscalademo.*
+
+/**
+ * This [[System]] is used to create a new ball into the [[World]]. If the mouse pointer is in the area of another ball,
+ * no ball will be added.
+ */
+class BallCreationSystem extends EmptySystem with ECScalaDSL {
+
+  override def shouldRun: Boolean = MouseState.clicked && !PlayState.playing
+
+  override def update(deltaTime: DeltaTime, world: World): Unit = {
+    val canBeCreated = world.getView[Position &: Circle &: CNil] map (_._2) forall { cl =>
+      val Position(point) &: Circle(radius, _) &: CNil = cl
+      !isOverlapped(point, radius)
+    }
+    if (canBeCreated) {
+      world hasAn entity withComponents {
+        Position(MouseState.coordinates) &:
+          Circle(StartingState.startingRadius, StartingState.startingColor) &:
+          Velocity(Vector(0, 0)) &:
+          Mass(StartingState.startingMass)
```

Comment:
Did scalafmt force you to indent the following lines? I think this would be nicer
```suggestion
        Position(MouseState.coordinates) &:
        Circle(StartingState.startingRadius, StartingState.startingColor) &:
        Velocity(Vector(0, 0)) &:
        Mass(StartingState.startingMass)
```

Reply:
I agree with you but scalafmt forces me to use that formatting...